src/mixer/basetrackplayer.cpp

```diff
@@ -792,6 +805,38 @@ void BaseTrackPlayerImpl::slotSetTrackColor(const mixxx::RgbColor::optional_t& c
     m_pTrackColor->forceSet(trackColorToDouble(color));
 }
 
+void BaseTrackPlayerImpl::slotTrackColorPrev(double v) {
+    if (!m_pLoadedTrack) {
+        return;
+    }
+
+    if (v > 0) {
+        ColorPaletteSettings colorPaletteSettings(m_pConfig);
+        ColorPalette colorPalette = colorPaletteSettings.getTrackColorPalette();
+        mixxx::RgbColor::optional_t color = m_pLoadedTrack->getColor();
+        color = colorPalette.previousColor(color);
+        m_pTrackColor->setAndConfirm(trackColorToDouble(color));
+        m_pLoadedTrack->setColor(color);
+    }
+}
+
+void BaseTrackPlayerImpl::slotTrackColorNext(double v) {
+    if (!m_pLoadedTrack) {
+        return;
+    }
+
+    if (v > 0) {
+        ColorPaletteSettings colorPaletteSettings(m_pConfig);
+        ColorPalette colorPalette = colorPaletteSettings.getTrackColorPalette();
+        mixxx::RgbColor::optional_t color = m_pLoadedTrack->getColor();
+
+        color = colorPalette.nextColor(color);
+
+        m_pTrackColor->setAndConfirm(trackColorToDouble(color));
+        m_pLoadedTrack->setColor(color);
```

Comment:
I think (!) `m_pLoadedTrack->setColor(color);` is sufficient since the track emits `colorUpdated(color)` which is caught here by `slotSetTrackColor()`, which in turn force-sets the control.

(I'm aware it's your solution is also used in `slotTrackColorChangeRequest()`, maybe fo a reason?)

Reply:
That's right !
It's checked in `forceSet()`, in `slotSetTrackColor()`.
That was kind of hidden, thanks for the event clarification.
Removing the extra check.